gui/src/2D/advancedDynamicTexture.ts

```diff
@@ -624,9 +656,51 @@ export class AdvancedDynamicTexture extends DynamicTexture {
             }
         });
 
+        this.onClipboardObservable.add((pi, state) => {

```

Comment:
Same here. the adt must be loosely coupled. So this is the responsability of the input text to register to the onClpiboardObservable and deals with data